### ISSUE TITLE
Can not upgrade to Habanero Red (0.10.2) on Azure. 

### ISSUE THREAD

**Issue by cfaagaard** [nightscout/cgm-remote-monitor#3107]
There is no node.js version 8.9.1 on azure app service. And then I can not upgrade.
I have version 8.9.0 on my azure app service. Se below:
{"nodejs":[{"version":"0.10.40","npm":"1.4.28"},{"version":"0.10.5","npm":"1.2.18"},{"version":"0.12.6","npm":"2.11.2"},{"version":"0.6.20","npm":"1.1.37"},{"version":"0.8.28","npm":"1.2.30"},{"version":"4.8.4","npm":"2.15.11"},{"version":"6.10.0","npm":"3.10.10"},{"version":"6.11.1","npm":"3.10.10"},{"version":"6.11.2","npm":"3.10.10"},{"version":"6.11.5","npm":"3.10.10"},{"version":"6.3.0","npm":"3.10.3"},{"version":"6.5.0","npm":"3.10.3"},{"version":"6.7.0","npm":"3.10.3"},{"version":"6.9.1","npm":"3.10.8"},{"version":"6.9.4","npm":"3.10.10"},{"version":"6.9.5","npm":"3.10.10"},{"version":"7.10.1","npm":"4.2.0"},{"version":"8.0.0","npm":"5.0.0"},{"version":"8.1.4","npm":"5.0.3"},{"version":"8.4.0","npm":"5.3.0"},{"version":"8.5.0","npm":"5.3.0"},{"version":"8.7.0","npm":"5.4.2"},{"version":"8.8.1","npm":"5.4.2"},{"version":"8.9.0","npm":"5.5.1"}]}

The node.js version on azure can be found here: https://MYSITE.scm.azurewebsites.net/api/diagnostics/runtime (replace "mysite")

Can not seem to find any info how to install version 8.9.1 on azure.

**Comment by bordeb** [comment]
Same issue here. Constant error in Azure when trying to sync, although when I visit the site it displays version 0.10.2-release-20171201 
Log error from Azure > Deployment options > Sync: 
_Node.js versions available on the platform are: 0.6.20, 0.8.2, 0.8.19, 0.8.26, 0.8.27, 0.8.28, 0.10.5, 0.10.18, 0.10.21, 0.10.24, 0.10.26, 0.10.28, 0.10.29, 0.10.31, 0.10.32, 0.10.40, 0.12.0, 0.12.2, 0.12.3, 0.12.6, 4.0.0, 4.1.0, 4.1.2, 4.2.1, 4.2.2, 4.2.3, 4.2.4, 4.3.0, 4.3.2, 4.4.0, 4.4.1, 4.4.6, 4.4.7, 4.5.0, 4.6.0, 4.6.1, 4.8.4, 5.0.0, 5.1.1, 5.3.0, 5.4.0, 5.5.0, 5.6.0, 5.7.0, 5.7.1, 5.8.0, 5.9.1, 6.0.0, 6.1.0, 6.2.2, 6.3.0, 6.5.0, 6.6.0, 6.7.0, 6.9.0, 6.9.1, 6.9.2, 6.9.4, 6.9.5, 6.10.0, 6.10.3, 6.11.1, 6.11.2, 6.11.5, 7.0.0, 7.1.0, 7.2.0, 7.3.0, 7.4.0, 7.5.0, 7.6.0, 7.7.0, 7.7.4, 7.10.0, 7.10.1, 8.0.0, 8.1.4, 8.4.0, 8.5.0, 8.7.0, 8.8.0, 8.8.1, 8.9.0.
No available node.js version matches application's version constraint of '8.9.1'. Use package.json to choose one of the available versions.
An error has occurred during web site deployment._

**Comment by christerjensen** [comment]
It is resolved now - try again.

**Comment by bordeb** [comment]
TNX very much! I can confirm, successful update on Azure after merging/"updating" my github repo today. Nevertheless, I still see the same version info (0.10.2-release-20171201) on Nightscout portal as it was yesterday prior to this latest "bugfix" changes.

**Comment by cfaagaard** [comment]
hmmmm..... I am not there yet.... Now I get this error:
Command: bash deploy.sh
Handling node.js deployment.
KuduSync.NET from: 'D:\home\site\repository' to: 'D:\home\site\wwwroot'
Copying file: 'scm-commit-id.json'
Using start-up script server.js from package.json.
Node.js versions available on the platform are: 0.6.20, 0.8.2, 0.8.19, 0.8.26, 0.8.27, 0.8.28, 0.10.5, 0.10.18, 0.10.21, 0.10.24, 0.10.26, 0.10.28, 0.10.29, 0.10.31, 0.10.32, 0.10.40, 0.12.0, 0.12.2, 0.12.3, 0.12.6, 4.0.0, 4.1.0, 4.1.2, 4.2.1, 4.2.2, 4.2.3, 4.2.4, 4.3.0, 4.3.2, 4.4.0, 4.4.1, 4.4.6, 4.4.7, 4.5.0, 4.6.0, 4.6.1, 4.8.4, 5.0.0, 5.1.1, 5.3.0, 5.4.0, 5.5.0, 5.6.0, 5.7.0, 5.7.1, 5.8.0, 5.9.1, 6.0.0, 6.1.0, 6.2.2, 6.3.0, 6.5.0, 6.6.0, 6.7.0, 6.9.0, 6.9.1, 6.9.2, 6.9.4, 6.9.5, 6.10.0, 6.10.3, 6.11.1, 6.11.2, 6.11.5, 7.0.0, 7.1.0, 7.2.0, 7.3.0, 7.4.0, 7.5.0, 7.6.0, 7.7.0, 7.7.4, 7.10.0, 7.10.1, 8.0.0, 8.1.4, 8.4.0, 8.5.0, 8.7.0, 8.8.0, 8.8.1, 8.9.0.
Selected node.js version 8.9.0. Use package.json file to choose a different version.
Selected npm version 5.3.0
Updating iisnode.yml at D:\home\site\wwwroot\iisnode.yml
npm WARN invalid config scripts-prepend-node-path="--production"
npm ERR! path D:\home\site\wwwroot\node_modules\fsevents\node_modules\getpass\node_modules
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall scandir
npm ERR! Error: EPERM: operation not permitted, scandir 'D:\home\site\wwwroot\node_modules\fsevents\node_modules\getpass\node_modules'
npm ERR!  { Error: EPERM: operation not permitted, scandir 'D:\home\site\wwwroot\node_modules\fsevents\node_modules\getpass\node_modules'
npm ERR!   stack: 'Error: EPERM: operation not permitted, scandir \'D:\\home\\site\\wwwroot\\node_modules\\fsevents\\node_modules\\getpass\\node_modules\'',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'scandir',
npm ERR!   path: 'D:\\home\\site\\wwwroot\\node_modules\\fsevents\\node_modules\\getpass\\node_modules' }
An error has occurred during web site deployment.
npm failed
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.

**Comment by bordeb** [comment]
Maybe just try one, two, three... more times to manually push the Sync button in Azure Deployment Section.
It sounds as a joke, I understand, but I am really not joking at all. :-) From my short experience (couple of months), many times after successful pull/merge/update of your personal github repo, unfortunately it takes several times to manually Sync until success (no errors) in Azure. I have never figured it out whats the reason. For example today, one of my Nightscouts portal updated at first try but other it took at least 5 times manual Sync until success. If someone knows what is the catch I would really like to hear it.

**Comment by cfaagaard** [comment]
ok. I will give it a try. (or 10 tries :-) )

**Comment by sulkaharo** [comment]
Looks like you’ve hit a known Azure bug. To fix, you need to disconnect Github link from your Azure account and re-link, then deploy the site again. (Or at least this did work in previous cases.)

**Comment by cfaagaard** [comment]
ok. thanks. I will try this.

**Comment by cfaagaard** [comment]
Tried everything.... nothing worked...... then suddenly everything worked..... hmmmmmm..
I can confirm that it is now possible to upgrade Habanero Red (0.10.2) on Azure after the fix. (the problems later was totally Azure related)

**Comment by tepidjuice** [comment]
I had exactly @cfaagaard 's problem. Should this go in the docs somewhere?

**Comment by sulkaharo** [comment]
@tepidjuice I think the support team is more inclined to start pushing users to Heroku, which has none of these issues in the first place...

**Comment by bordeb** [comment]
@sulkaharo thank you on such a valuable insider info! I have not achieved successful implementation at first on heroku couple of months ago so I have left that aside and just used azure, but knowing your info now I am going to retry to accomplish it also on heroku.
@cfaagaard thank you on mentioning me in your thanks sentence, but all the credits go to @nightscout team and  @sulkaharo for their astonishing work. I was just nearby and tried to help providing info on my strange experience with azure apps. :-) We all here need to support each other, especially because t1d is so much easier to manage in a good company.

**Comment by Cagier** [comment]
I'm still getting this error repeatedly, despite multiple synchs.  The repo is up to date and the node.js setting is 8.9.0

`Command: bash deploy.sh
Handling node.js deployment.
KuduSync.NET from: 'D:\home\site\repository' to: 'D:\home\site\wwwroot'
Copying file: 'scm-commit-id.json'
Using start-up script server.js from package.json.
Node.js versions available on the platform are: 0.6.20, 0.8.2, 0.8.19, 0.8.26, 0.8.27, 0.8.28, 0.10.5, 0.10.18, 0.10.21, 0.10.24, 0.10.26, 0.10.28, 0.10.29, 0.10.31, 0.10.32, 0.10.40, 0.12.0, 0.12.2, 0.12.3, 0.12.6, 4.0.0, 4.1.0, 4.1.2, 4.2.1, 4.2.2, 4.2.3, 4.2.4, 4.3.0, 4.3.2, 4.4.0, 4.4.1, 4.4.6, 4.4.7, 4.5.0, 4.6.0, 4.6.1, 4.8.4, 5.0.0, 5.1.1, 5.3.0, 5.4.0, 5.5.0, 5.6.0, 5.7.0, 5.7.1, 5.8.0, 5.9.1, 6.0.0, 6.1.0, 6.2.2, 6.3.0, 6.5.0, 6.6.0, 6.7.0, 6.9.0, 6.9.1, 6.9.2, 6.9.4, 6.9.5, 6.10.0, 6.10.3, 6.11.1, 6.11.2, 6.11.5, 7.0.0, 7.1.0, 7.2.0, 7.3.0, 7.4.0, 7.5.0, 7.6.0, 7.7.0, 7.7.4, 7.10.0, 7.10.1, 8.0.0, 8.1.4, 8.4.0, 8.5.0, 8.7.0, 8.8.0, 8.8.1, 8.9.0.
Selected node.js version 8.9.0. Use package.json file to choose a different version.
Selected npm version 5.3.0
Updating iisnode.yml at D:\home\site\wwwroot\iisnode.yml
npm WARN invalid config scripts-prepend-node-path="--production"
npm ERR! path D:\home\site\wwwroot\node_modules\fsevents\node_modules
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall scandir
npm ERR! Error: EPERM: operation not permitted, scandir 'D:\home\site\wwwroot\node_modules\fsevents\node_modules'
npm ERR!  { Error: EPERM: operation not permitted, scandir 'D:\home\site\wwwroot\node_modules\fsevents\node_modules'
npm ERR!   stack: 'Error: EPERM: operation not permitted, scandir \'D:\\home\\site\\wwwroot\\node_modules\\fsevents\\node_modules\'',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'scandir',
npm ERR!   path: 'D:\\home\\site\\wwwroot\\node_modules\\fsevents\\node_modules' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     D:\local\AppData\npm-cache\_logs\2017-12-09T14_06_21_718Z-debug.log
An error has occurred during web site deployment.
npm failed
npm WARN invalid config scripts-prepend-node-path="--production"\r\nnpm ERR! path D:\home\site\wwwroot\node_modules\fsevents\node_modules\r\nnpm ERR! code EPERM\r\nnpm ERR! errno -4048\r\nnpm ERR! syscall scandir\r\nnpm ERR! Error: EPERM: operation not permitted, scandir 'D:\home\site\wwwroot\node_modules\fsevents\node_modules'\r\nnpm ERR!  { Error: EPERM: operation not permitted, scandir 'D:\home\site\wwwroot\node_modules\fsevents\node_modules'\r\nnpm ERR!   stack: 'Error: EPERM: operation not permitted, scandir \'D:\\home\\site\\wwwroot\\node_modules\\fsevents\\node_modules\'',\r\nnpm ERR!   errno: -4048,\r\nnpm ERR!   code: 'EPERM',\r\nnpm ERR!   syscall: 'scandir',\r\nnpm ERR!   path: 'D:\\home\\site\\wwwroot\\node_modules\\fsevents\\node_modules' }\r\nnpm ERR! \r\nnpm ERR! Please try running this command again as root/Administrator.\r\n\r\nnpm ERR! A complete log of this run can be found in:\r\nnpm ERR!     D:\local\AppData\npm-cache\_logs\2017-12-09T14_06_21_718Z-debug.log\r\nD:\Program Files (x86)\SiteExtensions\Kudu\69.61204.3166\bin\Scripts\starter.cmd bash deploy.sh`

**Comment by helenday** [comment]
I disconnected from github as advised above and then resynched.  I got different errors after the first run and then reverted to the errors above with subsequent synchs.  Possibly the log from the first deployment attempt may shed some more light on the issue...

Command: bash deploy.sh
Handling node.js deployment.
KuduSync.NET from: 'D:\home\site\repository' to: 'D:\home\site\wwwroot'
Copying file: 'scm-commit-id.json'
Using start-up script server.js from package.json.
Node.js versions available on the platform are: 0.6.20, 0.8.2, 0.8.19, 0.8.26, 0.8.27, 0.8.28, 0.10.5, 0.10.18, 0.10.21, 0.10.24, 0.10.26, 0.10.28, 0.10.29, 0.10.31, 0.10.32, 0.10.40, 0.12.0, 0.12.2, 0.12.3, 0.12.6, 4.0.0, 4.1.0, 4.1.2, 4.2.1, 4.2.2, 4.2.3, 4.2.4, 4.3.0, 4.3.2, 4.4.0, 4.4.1, 4.4.6, 4.4.7, 4.5.0, 4.6.0, 4.6.1, 4.8.4, 5.0.0, 5.1.1, 5.3.0, 5.4.0, 5.5.0, 5.6.0, 5.7.0, 5.7.1, 5.8.0, 5.9.1, 6.0.0, 6.1.0, 6.2.2, 6.3.0, 6.5.0, 6.6.0, 6.7.0, 6.9.0, 6.9.1, 6.9.2, 6.9.4, 6.9.5, 6.10.0, 6.10.3, 6.11.1, 6.11.2, 6.11.5, 7.0.0, 7.1.0, 7.2.0, 7.3.0, 7.4.0, 7.5.0, 7.6.0, 7.7.0, 7.7.4, 7.10.0, 7.10.1, 8.0.0, 8.1.4, 8.4.0, 8.5.0, 8.7.0, 8.8.0, 8.8.1, 8.9.0.
Selected node.js version 8.9.0. Use package.json file to choose a different version.
Selected npm version 5.3.0
Updating iisnode.yml at D:\home\site\wwwroot\iisnode.yml
npm WARN invalid config scripts-prepend-node-path="--production"
npm WARN lifecycle The node binary used for scripts is D:\Program Files (x86)\nodejs\0.10.28\node.exe but npm is using D:\Program Files (x86)\nodejs\8.9.0\node.exe itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> Nightscout@0.10.2-release-20171201 postinstall D:\home\site\wwwroot
> webpack --config webpack.config.js && npm run-script update-buster


D:\home\site\wwwroot\node_modules\webpack\bin\webpack.js:152
yargs.parse(process.argv.slice(2), (err, argv, output) => {
                                                        ^
SyntaxError: Unexpected token >
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:935:3
npm ERR! code ELIFECYCLE
npm ERR! errno 8
npm ERR! Nightscout@0.10.2-release-20171201 postinstall: `webpack --config webpack.config.js && npm run-script update-buster`
npm ERR! Exit status 8
